packages/light-apps/src/AddAccount/util.ts

```diff
@@ -22,22 +47,27 @@ export function generateAddressFromMnemonic (keyring: Keyring, mnemonic: string)
 /**
  * Validate user inputs
  */
-export function validate (values: UserInput): Either<UserInputError, UserInput> {
+export function validateMeta (values: UserInput, step: string, whichAccount: string): Either<UserInputError, UserInput> {
   const errors = {} as UserInputError;
 
-  (['name', 'password', 'rewritePhrase'] as (Exclude<keyof UserInput, 'tags'>)[])
+  // @ts-ignore
+  values.tags = values.tags.map(tag => tag.toLowerCase());
+
+  if (step === 'meta') {
+    (['name', 'password'] as (Exclude<keyof UserInput, 'tags'>)[])
     .filter((key) => !values[key])
     .forEach((key) => {
       errors[key] = `Field "${key}" cannot be empty`;
     });
-
-  if (values.mnemonic !== values.rewritePhrase) {
-    errors.rewritePhrase = 'Mnemonic does not match rewrite';
+    debugger;
+    // if creating stash tag should be stash
+    if (whichAccount === 'stash' && !values.tags.includes('stash')) {
+      errors.tags = `The stash/controller difference is not enforced onchain. To avoid confusion, we require that you tag this account as a "${whichAccount}".`;
```

Comment:
Just a thought: Instead of showing this error message, it'd be simpler to automatically add tags to these accounts, and don't even show the Input.

Reply:
+1, I remember going down this way at first and abandoning it because it was too hairy in some way or another, but I think I have a better idea how to do this now.